fossunited/api/dashboard.py

```diff
@@ -13,8 +13,18 @@ def get_event(name: str) -> dict:
 
 
 @frappe.whitelist(allow_guest=True)
-def get_event_from_permalink(permalink: str) -> dict:
-    return frappe.db.get_value("FOSS Chapter Event", {"event_permalink": permalink}, ["*"], as_dict=1)
+def get_event_from_permalink(permalink: str, fields: list) -> dict:
+    return frappe.db.get_value("FOSS Chapter Event", {"event_permalink": permalink}, fields, as_dict=1)
+
+
+@frappe.whitelist(allow_guest=True)
+def get_cfp_details_from_eventname(eventname: str, fields: list) -> dict:
+    return frappe.db.get_value("FOSS Event CFP", {"event_name": eventname}, fields, as_dict=1)
+
+
+@frappe.whitelist(allow_guest=True)
+def get_proposal_list_from_eventname(eventname: str, fields: list) -> dict:
+    return frappe.get_all("FOSS Event CFP Submission", filters={"event_name": eventname}, fields=fields, page_length=99999, order_by="talk_title")
```

Comment:
This method can be refactored such that the details of proposal + likes on it can be transferred together. 

+ also add docstrings please

- Get all the proposals by get all
- For each proposal, get the likes count
- Stroke the likes data in another "like_data" key or something for the "proposal" object.
- Return the modified all proposals.

There's a neat function in fossunited.fossunited.utils called as `get_doc_likes`, which can be useful here.

```
from fossunited.fossunited.utils import get_doc_likes

def get_proposals_list(event_id: str, fields: list ) -> list:
    '''
    Returns the list of proposals for the given event, along with their likes data.

    Args:
        event_id (str): The event id for which the proposals are to be fetched.
        fields (list): The fields to be fetched from the proposals.

    Returns:
        list: The list of proposals for the given event.
    '''

    proposals = frappe.db.get_all(
        "FOSS Event CFP Submissions",
        {"event": event_id},
        fields=fields,
        page_length=9999,
        order_by="creation desc",
    )

    for proposal in proposals:
        proposal["likes"] = get_doc_likes("FOSS Event CFP Submission" , proposal["name"])
    
    return proposals

```

Please test/make changes as per requirements

Reply:
Originally I had used createListResource to fetch proposals - and `_liked_by` was not getting fetched in that. Hence I had made a separate function to fetch proposalLikes. 

However on a final test - I understood that createListResource won't work on public pages as it has a get_list invoke - and it's not whitelisted. 
So, as mentioned in the description, had to make quick whitelisted APIs that fetches the CFP and proposal details. 

I didn't append the proposalLikes fetching function here as later on there is a scope to build on top of the individual method - when implementing - likes feature on proposal cards (as I've mentioned in Additional Context). I have used the `get_doc_likes()` method here too. 

Also keeping the `proposal fetch` function separate was intentional - as that also has scope to be reused as it is - if we ever implement a feature of adding proposals submitted by a user on profile, as referred in #529    

However, now thinking in this angle - I came across frappe `[toggle_like](https://github.com/frappe/frappe/blob/develop/frappe/desk/like.py)` api which can be used here. Hence would update code accordingly.

Reply:
Let me know if it still need to be made one API

Reply:
I would rather that the api endpoints are really specific rather than very generic, which may accidentally leak the info. 

For example, in this method, I don't think we would want to fetch email fields. But any signed in user can test this api and pass `email` as a field and fetch it for all proposals of an event. Ofcourse there's the requirement of knowing event id, but it's not impossible. 

On other hand, let's just fetch likes data and proposal details in a single api endpoint. Even if it isn't used somewhere else, it's data associated with that proposal.

Reply:
Let's remove fields as an argument that you can pass.

Let's define the fields we want to fetch in the function itself, and make sure that we are not fetching sensitive datafields.

Reply:
I agree that the API end points need to be specific, but even with the said implementation that you mentioned - someone can fetch emails with the API right, as fields aren't fixed defined ? 

Unless the doctype itself is protected with permissive settings ?

Reply:
Sure, will make the changes

Reply:
There's similar code in reviewer.py, that may be helpful

Reply:
This is updated